core/src/services/ghaa/backend.rs

```
@@ -0,0 +1,279 @@
+        am.set_scheme(Scheme::Ghaa)
+            .set_native_capability(Capability {
+                stat: true,
+                read: true,
+                delete: true,
+                list: true,
+                list_with_limit: true,
+                ..Default::default()
+            });
+        am
+    }
+
+    async fn read(&self, path: &str, _: OpRead) -> Result<(RpRead, Self::Reader)> {
+        let artifact_id = self
+            .core
+            .get_artifact_id_by_path(path)
+            .await?
+            .ok_or(Error::new(
+                ErrorKind::NotFound,
+                &format!("path not found: {}", path),
+            ))?;
+        let resp = self.core.download_artifact(artifact_id.as_str()).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => Ok((RpRead::new(), resp.into_body())),
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn stat(&self, path: &str, _: OpStat) -> Result<RpStat> {
+        let artifact_id = self
+            .core
+            .get_artifact_id_by_path(path)
+            .await?
+            .ok_or(Error::new(
+                ErrorKind::NotFound,
+                &format!("path not found: {}", path),
+            ))?;
+        let resp = self.core.stat_artifact(artifact_id.as_str()).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => {
+                let slc = resp.into_body().bytes().await?;
+                let parsed_body: GhaaArtifact =
+                    serde_json::from_slice(&slc).map_err(new_json_deserialize_error)?;
+
+                let mut meta = Metadata::new(EntryMode::FILE);
+                meta.set_content_length(parsed_body.size_in_bytes);
+                meta.set_last_modified(
+                    parse_datetime_from_rfc3339(&parsed_body.updated_at)
+                        .map(|dt| dt.with_timezone(&Utc))
+                        .unwrap(),
```

Please return this error instead.